va/va.go

```diff
@@ -374,7 +384,14 @@ func (va VAImpl) validateTLSALPN01(task *vaTask) *core.ValidationRecord {
 	leafCert := certs[0]
 
 	// Verify SNI - certificate returned must be issued only for the domain we are verifying.
-	if len(leafCert.DNSNames) != 1 || !strings.EqualFold(leafCert.DNSNames[0], task.Identifier) {
+	namematch := false
+	switch task.IdentifierType {
```

Comment:
Above you used an `if`/`else` to switch on identifier type, here you're using `switch`. Can you update one to be consistent with the other?